pkg/manager/member/tikv_scaler.go

```diff
@@ -247,9 +248,14 @@ func (s *tikvScaler) preCheckUpStores(tc *v1alpha1.TidbCluster, podName string)
 	if err != nil {
 		return false, fmt.Errorf("failed to get stores info in TidbCluster %s/%s", tc.GetNamespace(), tc.GetName())
 	}
+	// filter out TiFlash
 	for _, store := range storesInfo.Stores {
-		if store.Store != nil && store.Store.StateName == v1alpha1.TiKVStateUp {
-			upNumber++
+		if store.Store != nil {
+			storePodName := strings.Split(store.Store.Address, ".")[0]
+			podKind := strings.Split(storePodName, "-")
```

Comment:
Do we need to check the `len(podKind)`?

Reply:
The pod name is generated in the format "xxx-xxx-tikv(component)-0(ordinals)", the last but one is fixed. Is there any corner case on it?

Reply:
Oh, I see. There might be some instances from outside K8S clusters.

Reply:
Fixed. I have updated UT to cover store named "outsidetikv".